src/neural/cuda/network_cuda.cc

```diff
@@ -49,24 +49,25 @@ using namespace cudnn_backend;
 template <typename DataType>
 class CudaNetwork;
 
-static size_t getMaxAttentionHeadSize(const LegacyWeights& weights, int N) {
-  const size_t embedding_op_size = weights.ip_pol_b.size();
-  const size_t policy_d_model = weights.ip2_pol_b.size();
-  assert(policy_d_model == weights.ip3_pol_b.size());
+static size_t getMaxAttentionHeadSize(const MultiHeadWeights& weights, int N) {
```

Comment:
I think a better approach is to pass a `MultiHeadWeights::PolicyHead&` instead of `LegacyWeights&`.

Reply:
Done